src/common/config_opts.h

```diff
@@ -188,6 +188,7 @@ OPTION(ms_rwthread_stack_bytes, OPT_U64, 1024 << 10)
 OPTION(ms_tcp_read_timeout, OPT_U64, 900)
 OPTION(ms_pq_max_tokens_per_priority, OPT_U64, 16777216)
 OPTION(ms_pq_min_cost, OPT_U64, 65536)
+OPTION(ms_immediate_fail_on_connection_refused, OPT_BOOL, true) // mark osd down immediately after receiving ECONNREFUSED during (re)connection
```

Comment:
I think this should be an osd option, osd_fast_fail_on_connection_refused, and the conditional should go in the OSD's ms_refused callback